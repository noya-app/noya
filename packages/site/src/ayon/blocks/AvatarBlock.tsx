import { applyCommonProps } from '@noya-design-system/protocol';
import { BlockDefinition } from 'noya-state';
import { isExternalUrl } from 'noya-utils';
import { getBlockClassName } from '../tailwind/tailwind';
import { getParameters } from '../utils/getMappedParameters';
import { isApproximatelySquare, isWithinRectRange } from './score';
import { avatarSymbolId } from './symbolIds';
import { avatarSymbol } from './symbols';

const AVATAR_SIZES = ['2xs', 'xs', 'sm', 'md', 'lg', 'xl', '2xl'];

export const AvatarBlock: BlockDefinition = {
  symbol: avatarSymbol,
  infer: ({ frame, blockText }) =>
    isWithinRectRange({
      rect: frame,
      minWidth: 30,
      minHeight: 30,
      maxWidth: 120,
      maxHeight: 120,
    }) && isApproximatelySquare(frame, 0.2)
      ? 0.8
      : 0,
  render: ({ h, Components: { [avatarSymbolId]: Avatar } }, props) => {
    const parameters = getParameters(props.blockParameters);
    const content = props.blockText ?? '';
    const src = isExternalUrl(content) ? content : undefined;
    const name = !src ? content : undefined;
    const hashtags = Object.keys(parameters);

    // Size up in 14px increments
    const size = props.frame
      ? AVATAR_SIZES[
          Math.min(
            Math.floor(Math.max(props.frame.width, props.frame.height) / 14),
            AVATAR_SIZES.length - 1,
          )
        ]
      : 'md';

    const sizePx = props.frame
      ? Math.min(props.frame.width, props.frame.height)
      : undefined;

    const widthHashtag = hashtags.find((hashtag) => hashtag.startsWith('w-'));
    const widthMatch = widthHashtag ? widthHashtag.match(/w-(\d+)/) : undefined;
    const width = widthMatch ? Number(widthMatch[1]) : undefined;
    const tailwindSize = width ? `${width * 0.25}rem` : undefined;

    return h(Avatar, {
      ...applyCommonProps(props),
      size,
      name,
      src,
      style: {
        aspectRatio: '1',
        ...(sizePx && {
          width: `${sizePx}px`,
          height: `${sizePx}px`,
        }),
        ...(tailwindSize && {
          width: tailwindSize,
          height: tailwindSize,
        }),
      },
      className: getBlockClassName(hashtags),
    });
  },
};
