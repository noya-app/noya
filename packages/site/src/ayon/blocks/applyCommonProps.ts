import { CommonProps } from '@noya-design-system/protocol';
import { BlockProps } from 'noya-state';

export function applyCommonProps(
  props: BlockProps,
): Pick<CommonProps, '_passthrough'> {
  return {
    _passthrough: {
      ...(props.dataSet && {
        key: props.dataSet.id,
        'data-noya-id': props.dataSet.id,
        'data-noya-parent-id': props.dataSet.parentId,
      }),
    },
  };
}
