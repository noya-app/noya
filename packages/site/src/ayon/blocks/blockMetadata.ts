import { Size } from 'noya-geometry';
import {
  avatarSymbolId,
  boxSymbolId,
  buttonSymbolId,
  cardSymbolId,
  checkboxSymbolId,
  featureItemSymbolId,
  headerBarSymbolId,
  heading1SymbolId,
  heading2SymbolId,
  heading3SymbolId,
  heading4SymbolId,
  heading5SymbolId,
  heading6SymbolId,
  heroSymbolId,
  heroWithImageSymbolId,
  iconSymbolId,
  imageSymbolId,
  inputSymbolId,
  linkSymbolId,
  radioSymbolId,
  selectSymbolId,
  sidebarSymbolId,
  signInSymbolId,
  switchSymbolId,
  tableSymbolId,
  tagSymbolId,
  textareaSymbolId,
  textSymbolId,
  tileCardSymbolId,
  writeSymbolId,
} from '../symbols/symbolIds';

export type BlockCategory = 'application' | 'marketing' | 'element';

export type PreferredOverride = {
  symbolId: string;
  blockText: string;
  resolvedBlockText?: string;
};

export type BlockMetadata = {
  name: string;
  category: BlockCategory;
  preferredSize: Size;
  preferredBlockText?: string;
  preferredResolvedBlockText?: string;
  preferredOverrides?: PreferredOverride[];
  hideInDocs?: boolean;
};

const headingMetadata: BlockMetadata = {
  name: 'Heading',
  category: 'element',
  preferredSize: { width: 700, height: 100 },
  preferredBlockText: 'All About Cats',
};

const headingNMetadata: BlockMetadata = {
  ...headingMetadata,
  hideInDocs: true,
};

// This is in a separate file so docs don't have to import Chakra components directly.
// Docs can import this synchronously and import(Ayon)
export const blockMetadata: Record<string, BlockMetadata> = {
  [avatarSymbolId]: {
    name: 'Avatar',
    category: 'element',
    preferredSize: { width: 64, height: 64 },
  },
  [buttonSymbolId]: {
    name: 'Button',
    category: 'element',
    preferredSize: { width: 200, height: 40 },
  },
  [linkSymbolId]: {
    name: 'Link',
    category: 'element',
    preferredSize: { width: 200, height: 40 },
  },
  [boxSymbolId]: {
    name: 'Box',
    category: 'element',
    preferredSize: { width: 200, height: 200 },
  },
  [heroSymbolId]: {
    name: 'Hero',
    category: 'marketing',
    preferredSize: { width: 600, height: 400 },
  },
  [heroWithImageSymbolId]: {
    name: 'Hero with Image',
    category: 'marketing',
    preferredSize: { width: 1280, height: 720 },
  },
  [tagSymbolId]: {
    name: 'Tag',
    category: 'element',
    preferredSize: { width: 140, height: 30 },
  },
  [cardSymbolId]: {
    name: 'Card',
    category: 'marketing',
    preferredSize: { width: 250, height: 300 },
    preferredOverrides: [
      {
        symbolId: imageSymbolId,
        blockText: 'landscape',
        resolvedBlockText:
          'https://images.unsplash.com/photo-1514917860136-ee8b88e8c9c9?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=300&ixid=MnwxfDB8MXxyYW5kb218MHx8bGFuZHNjYXBlLGJ1c3Rlcnx8fHx8fDE2Nzc2NDIxNDM&ixlib=rb-4.0.3&q=80&utm_campaign=api-credit&utm_medium=referral&utm_source=unsplash_source&w=300',
      },
    ],
  },
  [tileCardSymbolId]: {
    name: 'Tile Card',
    category: 'marketing',
    preferredSize: { width: 250, height: 250 },
  },
  [iconSymbolId]: {
    name: 'Icon',
    category: 'element',
    preferredSize: { width: 48, height: 48 },
  },
  [imageSymbolId]: {
    name: 'Image',
    category: 'element',
    preferredSize: { width: 300, height: 300 },
    preferredBlockText: 'landscape',
    preferredResolvedBlockText:
      'https://images.unsplash.com/photo-1514917860136-ee8b88e8c9c9?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=300&ixid=MnwxfDB8MXxyYW5kb218MHx8bGFuZHNjYXBlLGJ1c3Rlcnx8fHx8fDE2Nzc2NDIxNDM&ixlib=rb-4.0.3&q=80&utm_campaign=api-credit&utm_medium=referral&utm_source=unsplash_source&w=300',
  },
  [writeSymbolId]: {
    name: 'Write',
    category: 'element',
    preferredSize: { width: 300, height: 300 },
    preferredBlockText: 'Cats are great pets',
    preferredResolvedBlockText:
      'Cats are incredibly intelligent creatures with personalities all of their own. They make loyal and loving companions who can provide years of fun and joy. Cats come in all shapes, sizes, and colors and offer a variety of health benefits to their human owners. From their natural mouse-catching abilities to their playful antics, cats make terrific pet choices for a wide variety of households.',
  },
  [textSymbolId]: {
    name: 'Text',
    category: 'element',
    preferredSize: { width: 300, height: 150 },
    preferredBlockText:
      'Cats are great pets. They make loyal and loving companions who can provide years of fun and joy.',
  },
  [heading1SymbolId]: headingMetadata,
  [heading2SymbolId]: headingNMetadata,
  [heading3SymbolId]: headingNMetadata,
  [heading4SymbolId]: headingNMetadata,
  [heading5SymbolId]: headingNMetadata,
  [heading6SymbolId]: headingNMetadata,
  [signInSymbolId]: {
    name: 'Sign In',
    category: 'application',
    preferredSize: { width: 350, height: 250 },
  },
  [tableSymbolId]: {
    name: 'Table',
    category: 'application',
    preferredSize: { width: 600, height: 400 },
  },
  [sidebarSymbolId]: {
    name: 'Sidebar',
    category: 'application',
    preferredSize: { width: 250, height: 600 },
  },
  [headerBarSymbolId]: {
    name: 'Header Bar',
    category: 'application',
    preferredSize: { width: 800, height: 80 },
  },
  [checkboxSymbolId]: {
    name: 'Checkbox',
    category: 'element',
    preferredSize: { width: 300, height: 30 },
  },
  [radioSymbolId]: {
    name: 'Radio',
    category: 'element',
    preferredSize: { width: 300, height: 30 },
  },
  [selectSymbolId]: {
    name: 'Select',
    category: 'element',
    preferredSize: { width: 300, height: 40 },
  },
  [textareaSymbolId]: {
    name: 'Textarea',
    category: 'element',
    preferredSize: { width: 240, height: 120 },
  },
  [inputSymbolId]: {
    name: 'Input',
    category: 'element',
    preferredSize: { width: 240, height: 40 },
  },
  [switchSymbolId]: {
    name: 'Switch',
    category: 'element',
    preferredSize: { width: 50, height: 35 },
  },
  [featureItemSymbolId]: {
    name: 'Feature Item',
    category: 'marketing',
    preferredSize: { width: 400, height: 160 },
  },
};
