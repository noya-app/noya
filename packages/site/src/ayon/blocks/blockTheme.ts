// Blue
export const accentColor = {
  50: '#eff6ff',
  100: '#dbeafe',
  200: '#bfdbfe',
  300: '#93c5fd',
  400: '#60a5fa',
  500: '#3b82f6',
  600: '#2563eb',
  700: '#1d4ed8',
  800: '#1e40af',
  900: '#1e3a8a',
};

// Slate
export const neutralColor = {
  50: '#f8fafc',
  100: '#f1f5f9',
  200: '#e2e8f0',
  300: '#cbd5e1',
  400: '#94a3b8',
  500: '#64748b',
  600: '#475569',
  700: '#334155',
  800: '#1e293b',
  900: '#0f172a',
};

export const buttonColors = {
  light: {
    backgroundColor: '#f1f5f9',
    color: '#000',
  },
  lightDisabled: {
    backgroundColor: '#e2e8f0',
    color: '#000',
  },
  dark: {
    backgroundColor: '#475569',
    color: '#fff',
  },
  darkDisabled: {
    backgroundColor: '#8799b2',
    color: '#fff',
  },
};
