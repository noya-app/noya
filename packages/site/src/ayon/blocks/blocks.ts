import Sketch from 'noya-file-format';
import {
  heroButtonRowSymbol,
  heroHeadlineStackSymbol,
  heroSymbol,
} from '../symbols/composed/HeroSymbol';
import { avatarSymbol } from '../symbols/primitive/AvatarSymbol';
import { boxSymbol } from '../symbols/primitive/BoxSymbol';
import { buttonSymbol } from '../symbols/primitive/ButtonSymbol';
import { checkboxSymbol } from '../symbols/primitive/CheckboxSymbol';
import { tagSymbol } from '../symbols/primitive/TagSymbol';
import { heroSymbolV1Id } from '../symbols/symbolIds';

export const allSymbols = [
  avatarSymbol,
  buttonSymbol,
  boxSymbol,
  heroSymbol,
  heroHeadlineStackSymbol,
  heroButtonRowSymbol,
  tagSymbol,
  checkboxSymbol,
];

export const allInsertableSymbols = allSymbols.filter(
  (symbol) => !symbol.blockDefinition?.isPassthrough,
);

export const symbolMap: Record<string, Sketch.SymbolMaster> = {
  ...Object.fromEntries(allSymbols.map((symbol) => [symbol.symbolID, symbol])),
  // Add extra mappings here (e.g. heroV1 => heroV2)
  [heroSymbolV1Id]: heroSymbol,
};
