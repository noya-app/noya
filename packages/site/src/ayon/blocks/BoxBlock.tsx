import { BlockDefinition } from 'noya-state';
import { findLast } from 'noya-utils';
import { parseBlock } from '../parse';
import { applyCommonProps } from './applyCommonProps';
import { accentColor } from './blockTheme';
import { getBlockThemeColors } from './colors';
import { boxSymbolId } from './symbolIds';
import { boxSymbol } from './symbols';
import {
  getLastClassInGroup,
  parametersToTailwindStyle,
  tailwindBlockClasses,
} from './tailwind';

export const BoxBlock: BlockDefinition = {
  symbol: boxSymbol,
  parser: 'regular',
  infer: ({ frame, blockText }) => 0.1,
  hashtags: ['left', 'center', 'right', 'dark', ...tailwindBlockClasses],
  render: ({ h, Components: { [boxSymbolId]: Box } }, props) => {
    const { content, parameters } = parseBlock(props.blockText, 'regular');
    const hashtags = Object.keys(parameters);

    const bgClass = getLastClassInGroup('background', hashtags);
    const dynamicClass = bgClass
      ? typeof parameters[bgClass] === 'string'
      : undefined;

    const backgroundColor = parameters.dark
      ? getBlockThemeColors({ dark: true, accent: false }).backgroundColor
      : bgClass
      ? dynamicClass
        ? parameters[bgClass]
        : undefined
      : [content]
          .concat(hashtags)
          .find((value) => CSS.supports('color', `${value}`)) ??
        accentColor[50];

    const { justify, items } = simpleFlex(hashtags);

    const style = parametersToTailwindStyle({
      ...parameters,
      ...(justify && { [justify]: true }),
      ...(items && { [items]: true }),
    });

    return h(
      Box,
      {
        ...applyCommonProps(props),
        style: {
          display: 'flex',
          ...style,
          backgroundColor,
          ...(props.frame && {
            width: `${props.frame.width}px`,
            height: `${props.frame.height}px`,
          }),
        },
      },
      props.children,
    );
  },
};

// Simplify flexbox alignment by handling row vs. column automatically
function simpleFlex(hashtags: string[]) {
  const flexKey = findLast(hashtags, (value) =>
    /^(flex-row|flex-col)$/.test(value),
  );
  const alignmentKey = findLast(hashtags, (value) =>
    /^(left|center|right)$/.test(value),
  );

  let justify: string | undefined;
  let items: string | undefined;

  if (flexKey === 'flex-row') {
    switch (alignmentKey) {
      case 'left':
        justify = 'justify-start';
        break;
      case 'center':
        justify = 'justify-center';
        break;
      case 'right':
        justify = 'justify-end';
        break;
    }

    if (hashtags.includes('center')) {
      items = 'items-center';
    }
  } else if (flexKey === 'flex-col') {
    switch (alignmentKey) {
      case 'left':
        items = 'items-start';
        break;
      case 'center':
        items = 'items-center';
        break;
      case 'right':
        items = 'items-end';
        break;
    }

    if (hashtags.includes('center')) {
      justify = 'justify-center';
    }
  }

  return { justify, items };
}
