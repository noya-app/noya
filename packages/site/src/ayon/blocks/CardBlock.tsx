import { BlockDefinition } from 'noya-state';
import { BoxBlock } from './BoxBlock';
import { renderStack } from './render';
import { isWithinRectRange } from './score';
import { cardSymbol } from './symbols';

export const CardBlock: BlockDefinition = {
  editorVersion: 2,
  symbol: cardSymbol,
  hashtags: BoxBlock.hashtags,
  isComposedBlock: true,
  infer: ({ frame, blockText, siblingBlocks }) => {
    return Math.max(
      isWithinRectRange({
        rect: frame,
        minWidth: 200,
        minHeight: 250,
        maxWidth: 300,
        maxHeight: 400,
      })
        ? 1
        : 0,
      0.1,
    );
  },
  render: (env, props) => renderStack(env, { props, block: CardBlock }),
};
