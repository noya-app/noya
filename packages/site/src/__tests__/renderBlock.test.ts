import { DesignSystemDefinition } from '@noya-design-system/protocol';
import { readFileSync } from 'fs';
import {
  compile,
  createElement,
  createElementCode,
  format,
  print,
} from 'noya-compiler';
import Sketch from 'noya-file-format';
import { loadDesignSystem } from 'noya-module-loader';
import { SketchModel } from 'noya-sketch-model';
import { Layers, Overrides } from 'noya-state';
import path from 'path';
import { Blocks } from '../ayon/blocks/blocks';
import {
  buttonSymbolId,
  cardSymbolId,
  heroSymbolV2Id,
  iconSymbolId,
  imageSymbolId,
  linkSymbolId,
  textSymbolId,
} from '../ayon/blocks/symbolIds';

// Jest doesn't know how to import a text file, so we mock it
jest.mock('../../safelist.txt', () => {
  return {
    default: readFileSync(path.join(__dirname, '../../safelist.txt'), 'utf8'),
  };
});

let DesignSystem: DesignSystemDefinition;

beforeAll(async () => {
  DesignSystem = await loadDesignSystem('chakra');
});

function generate(symbol: Sketch.SymbolInstance) {
  return format(
    print(createElementCode(createElement({ Blocks, DesignSystem }, symbol)!)),
  );
}

describe('button', () => {
  test('default', () => {
    const symbol = SketchModel.symbolInstance({
      symbolID: buttonSymbolId,
      frame: SketchModel.rect({
        width: 100,
        height: 40,
      }),
    });

    expect(generate(symbol)).toMatchSnapshot();
  });
});

describe('text', () => {
  test('default', () => {
    const symbol = SketchModel.symbolInstance({
      symbolID: textSymbolId,
      blockText: 'Hello world #center #text-red-500',
      frame: SketchModel.rect({
        width: 100,
        height: 40,
      }),
    });

    expect(generate(symbol)).toMatchSnapshot();
  });

  test('escape jsx', () => {
    const symbol = SketchModel.symbolInstance({
      symbolID: textSymbolId,
      blockText: 'Hello "\'<{test}>\'" world',
      frame: SketchModel.rect({
        width: 100,
        height: 40,
      }),
    });

    expect(generate(symbol)).toMatchSnapshot();
  });
});

describe('link', () => {
  // TODO: Icon doesn't get added to exported code
  test('with icon', () => {
    const symbol = SketchModel.symbolInstance({
      symbolID: linkSymbolId,
      blockText: 'Test #icon-arrow-right',
      frame: SketchModel.rect({
        width: 100,
        height: 40,
      }),
    });

    expect(generate(symbol)).toMatchSnapshot();
  });
});

describe('icon', () => {
  test('default', () => {
    const symbol = SketchModel.symbolInstance({
      symbolID: iconSymbolId,
      frame: SketchModel.rect({
        width: 48,
        height: 48,
      }),
    });

    expect(generate(symbol)).toMatchSnapshot();
  });

  test('with color', () => {
    const symbol = SketchModel.symbolInstance({
      symbolID: iconSymbolId,
      blockText: '#fill-blue-400',
      frame: SketchModel.rect({
        width: 48,
        height: 48,
      }),
    });

    expect(generate(symbol)).toMatchSnapshot();
  });
});

describe('hero', () => {
  test('default', () => {
    const symbol = SketchModel.symbolInstance({
      symbolID: heroSymbolV2Id,
      frame: SketchModel.rect({
        width: 400,
        height: 400,
      }),
    });

    expect(generate(symbol)).toMatchSnapshot();
  });

  test('left aligned', () => {
    const symbol = SketchModel.symbolInstance({
      symbolID: heroSymbolV2Id,
      frame: SketchModel.rect({
        width: 400,
        height: 400,
      }),
      blockText: '#left',
    });

    expect(generate(symbol)).toMatchSnapshot();
  });

  test('with bg', () => {
    const symbol = SketchModel.symbolInstance({
      symbolID: heroSymbolV2Id,
      frame: SketchModel.rect({
        width: 400,
        height: 400,
      }),
      blockText: '#bg-blue-500',
    });

    expect(generate(symbol)).toMatchSnapshot();
  });
});

describe('card with no border radius on image', () => {
  test('default', () => {
    const symbol = SketchModel.symbolInstance({
      symbolID: cardSymbolId,
      frame: SketchModel.rect({
        width: 300,
        height: 400,
      }),
    });

    Blocks[symbol.symbolID].symbol.layers
      .filter(Layers.isSymbolInstance)
      .filter((layer) => layer.symbolID === imageSymbolId)
      .forEach((layer) => {
        symbol.overrideValues.push(
          SketchModel.overrideValue({
            overrideName: Overrides.encodeName(
              [layer.do_objectID],
              'resolvedBlockData',
            ),
            value: {
              symbolID: layer.symbolID,
              originalText: 'cat',
              resolvedText: 'https://placekitten.com/300/300',
            },
          }),
        );
        symbol.overrideValues.push(
          SketchModel.overrideValue({
            overrideName: Overrides.encodeName(
              [layer.do_objectID],
              'blockText',
            ),
            value: 'cat #rounded-none',
          }),
        );
      });

    expect(generate(symbol)).toMatchSnapshot();
  });
});

describe('generate file', () => {
  test('base', async () => {
    const artboard = SketchModel.artboard({
      frame: SketchModel.rect({
        width: 100,
        height: 100,
      }),
    });

    expect(
      await compile({
        artboard,
        Blocks,
        DesignSystem,
        target: 'standalone',
      }),
    ).toMatchSnapshot();
  });
});
