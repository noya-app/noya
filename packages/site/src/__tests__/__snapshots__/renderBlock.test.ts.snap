// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button default 1`] = `
"<Frame width={100} height={40}>
  <Button isDisabled={false} textColor=\\"#000\\" bgColor=\\"#f1f5f9\\" w=\\"100px\\">
    Submit
  </Button>
</Frame>;
"
`;

exports[`card with no border radius on image default 1`] = `
"<Frame width={300} height={400}>
  <Box
    display=\\"flex\\"
    w=\\"300px\\"
    h=\\"400px\\"
    className=\\"flex-col bg-white p-4 shadow rounded-md\\"
  >
    <Image
      src=\\"https://placekitten.com/300/300\\"
      minH={0}
      objectFit=\\"cover\\"
      className=\\"rounded-none flex-1\\"
    />
    <Box display=\\"flex\\" className=\\"mt-4\\">
      <Text fontSize=\\"2xl\\" flex=\\"1\\">
        News
      </Text>
    </Box>
    <Text className=\\"mt-1\\">
      Here you can explore the latest news and information.
    </Text>
  </Box>
</Frame>;
"
`;

exports[`element as prop default 1`] = `
"<Frame width={100} height={40}>
  <Box
    style={{ display: 'flex', alignItems: 'center', width: 100, height: 40 }}
  >
    <FormControlLabel
      control={<Checkbox checked={false} disabled={false} />}
      label=\\"Hello world\\"
    />
  </Box>
</Frame>;
"
`;

exports[`generate file chakra 1`] = `
Object {
  ".postcssrc": "{
  \\"plugins\\": {
    \\"tailwindcss\\": {}
  }
}",
  "App.tsx": "import { ChakraProvider, Box } from '@chakra-ui/react';
import * as React from 'react';
/**
 * To make your layout responsive, delete this Frame component and replace any
 * instance of it with your own layout components that use e.g. flexbox.
 */
function Frame(
  props: React.PropsWithChildren<
    Pick<React.CSSProperties, 'left' | 'top' | 'width' | 'height'>
  >
) {
  return (
    <Box
      style={{
        position: 'absolute',
        left: props.left,
        top: props.top,
        width: props.width,
        height: props.height,
      }}
    >
      {props.children}
    </Box>
  );
}
export function App() {
  return <ChakraProvider />;
}
",
  "README.md": "# My App

## Local Development

\`\`\`
npm install
npm run dev
\`\`\`

## Production Build

\`\`\`
npm run build
\`\`\`
",
  "index.css": "@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  font-synthesis: none;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}
",
  "index.html": "<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"utf-8\\" />
    <title>My App</title>
  </head>
  <body>
    <div id=\\"root\\"></div>
    <link href=\\"./index.css\\" rel=\\"stylesheet\\" />
    
    <script type=\\"module\\" src=\\"index.tsx\\"></script>
  </body>
</html>",
  "index.tsx": "import { createRoot } from \\"react-dom/client\\";
import * as React from 'react';

import { App } from \\"./App\\";

const rootElement = document.getElementById(\\"root\\");
const root = createRoot(rootElement!);

root.render(<App />);",
  "package.json": "{
  \\"name\\": \\"my-app\\",
  \\"version\\": \\"0.0.1\\",
  \\"scripts\\": {
    \\"dev\\": \\"parcel index.html --open\\",
    \\"build\\": \\"parcel build index.html\\"
  },
  \\"dependencies\\": {
    \\"react\\": \\"^18\\",
    \\"react-dom\\": \\"^18\\",
    \\"@chakra-ui/react\\": \\"^2\\",
    \\"@chakra-ui/system\\": \\"^2\\",
    \\"@emotion/react\\": \\"^11\\",
    \\"@emotion/styled\\": \\"^11\\",
    \\"framer-motion\\": \\"^10\\",
    \\"@chakra-ui/icons\\": \\"^2\\"
  },
  \\"devDependencies\\": {
    \\"autoprefixer\\": \\"10.4.14\\",
    \\"parcel\\": \\"^2.8.3\\",
    \\"postcss\\": \\"8.4.21\\",
    \\"tailwindcss\\": \\"3.2.7\\",
    \\"process\\": \\"^0.11.10\\",
    \\"@types/react\\": \\"^18\\",
    \\"@types/react-dom\\": \\"^18\\"
  }
}",
  "tailwind.config.js": "module.exports = {
  content: [\\"./*.{html,js,jsx,ts,tsx}\\"],
  theme: {
    extend: {},
  },
  variants: {},
  plugins: [],
};",
}
`;

exports[`generate file mui 1`] = `
Object {
  ".postcssrc": "{
  \\"plugins\\": {
    \\"tailwindcss\\": {}
  }
}",
  "App.tsx": "import { Box, FormControlLabel, Checkbox } from '@mui/material';
import '@fontsource/roboto/latin.css';
import * as React from 'react';
/**
 * To make your layout responsive, delete this Frame component and replace any
 * instance of it with your own layout components that use e.g. flexbox.
 */
function Frame(
  props: React.PropsWithChildren<
    Pick<React.CSSProperties, 'left' | 'top' | 'width' | 'height'>
  >
) {
  return (
    <Box
      style={{
        position: 'absolute',
        left: props.left,
        top: props.top,
        width: props.width,
        height: props.height,
      }}
    >
      {props.children}
    </Box>
  );
}
export function App() {
  return (
    <>
      <Frame width={100} height={40}>
        <Box
          style={{
            display: 'flex',
            alignItems: 'center',
            width: 100,
            height: 40,
          }}
        >
          <FormControlLabel
            control={<Checkbox checked={false} disabled={false} />}
            label=\\"Hello world\\"
          />
        </Box>
      </Frame>
    </>
  );
}
",
  "README.md": "# My App

## Local Development

\`\`\`
npm install
npm run dev
\`\`\`

## Production Build

\`\`\`
npm run build
\`\`\`
",
  "index.css": "@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  font-synthesis: none;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}
",
  "index.html": "<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"utf-8\\" />
    <title>My App</title>
  </head>
  <body>
    <div id=\\"root\\"></div>
    <link href=\\"./index.css\\" rel=\\"stylesheet\\" />
    
    <script type=\\"module\\" src=\\"index.tsx\\"></script>
  </body>
</html>",
  "index.tsx": "import { createRoot } from \\"react-dom/client\\";
import * as React from 'react';

import { App } from \\"./App\\";

const rootElement = document.getElementById(\\"root\\");
const root = createRoot(rootElement!);

root.render(<App />);",
  "package.json": "{
  \\"name\\": \\"my-app\\",
  \\"version\\": \\"0.0.1\\",
  \\"scripts\\": {
    \\"dev\\": \\"parcel index.html --open\\",
    \\"build\\": \\"parcel build index.html\\"
  },
  \\"dependencies\\": {
    \\"react\\": \\"^18\\",
    \\"react-dom\\": \\"^18\\",
    \\"@mui/material\\": \\"^5\\",
    \\"@emotion/react\\": \\"^11\\",
    \\"@emotion/styled\\": \\"^11\\",
    \\"@mui/icons-material\\": \\"^5\\",
    \\"@fontsource/roboto\\": \\"^5\\"
  },
  \\"devDependencies\\": {
    \\"autoprefixer\\": \\"10.4.14\\",
    \\"parcel\\": \\"^2.8.3\\",
    \\"postcss\\": \\"8.4.21\\",
    \\"tailwindcss\\": \\"3.2.7\\",
    \\"process\\": \\"^0.11.10\\",
    \\"@types/react\\": \\"^18\\",
    \\"@types/react-dom\\": \\"^18\\"
  }
}",
  "tailwind.config.js": "module.exports = {
  content: [\\"./*.{html,js,jsx,ts,tsx}\\"],
  theme: {
    extend: {},
  },
  variants: {},
  plugins: [],
};",
}
`;

exports[`hero default 1`] = `
"<Frame width={400} height={400}>
  <Box
    display=\\"flex\\"
    w=\\"400px\\"
    h=\\"400px\\"
    className=\\"flex-col bg-transparent p-4 gap-3 backdrop-blur items-center justify-center\\"
  >
    <Box display=\\"flex\\">
      <Text fontSize=\\"6xl\\" flex=\\"1\\" textAlign=\\"center\\">
        Create, iterate, inspire.
      </Text>
    </Box>
    <Box display=\\"flex\\" className=\\"mb-4\\">
      <Text fontSize=\\"2xl\\" flex=\\"1\\" textAlign=\\"center\\">
        Turn great ideas into new possibilities.
      </Text>
    </Box>
    <Box display=\\"flex\\" className=\\"bg-transparent flex-row items-center gap-6\\">
      <Button isDisabled={false} textColor=\\"#fff\\" bgColor=\\"#15803d\\">
        Get started
      </Button>
    </Box>
  </Box>
</Frame>;
"
`;

exports[`hero left aligned 1`] = `
"<Frame width={400} height={400}>
  <Box
    display=\\"flex\\"
    w=\\"400px\\"
    h=\\"400px\\"
    className=\\"flex-col bg-transparent p-4 gap-3 backdrop-blur items-start justify-center\\"
  >
    <Box display=\\"flex\\">
      <Text fontSize=\\"6xl\\" flex=\\"1\\" textAlign=\\"left\\">
        Create, iterate, inspire.
      </Text>
    </Box>
    <Box display=\\"flex\\" className=\\"mb-4\\">
      <Text fontSize=\\"2xl\\" flex=\\"1\\" textAlign=\\"left\\">
        Turn great ideas into new possibilities.
      </Text>
    </Box>
    <Box display=\\"flex\\" className=\\"bg-transparent flex-row items-center gap-6\\">
      <Button isDisabled={false} textColor=\\"#fff\\" bgColor=\\"#15803d\\">
        Get started
      </Button>
    </Box>
  </Box>
</Frame>;
"
`;

exports[`hero with bg 1`] = `
"<Frame width={400} height={400}>
  <Box
    display=\\"flex\\"
    w=\\"400px\\"
    h=\\"400px\\"
    className=\\"flex-col bg-blue-500 p-4 gap-3 backdrop-blur items-center justify-center\\"
  >
    <Box display=\\"flex\\">
      <Text fontSize=\\"6xl\\" flex=\\"1\\" textAlign=\\"center\\" className=\\"text-white\\">
        Create, iterate, inspire.
      </Text>
    </Box>
    <Box display=\\"flex\\" className=\\"mb-4\\">
      <Text fontSize=\\"2xl\\" flex=\\"1\\" textAlign=\\"center\\" className=\\"text-white\\">
        Turn great ideas into new possibilities.
      </Text>
    </Box>
    <Box display=\\"flex\\" className=\\"bg-transparent flex-row items-center gap-6\\">
      <Button
        isDisabled={false}
        textColor=\\"#000\\"
        bgColor=\\"#f1f5f9\\"
        className=\\"text-white\\"
      >
        Get started
      </Button>
    </Box>
  </Box>
</Frame>;
"
`;

exports[`icon default 1`] = `
"<Frame width={48} height={48}>
  <Image
    src=\\"https://api.iconify.design/material-symbols/menu.svg\\"
    objectFit=\\"contain\\"
    w=\\"100%\\"
    h=\\"100%\\"
  />
</Frame>;
"
`;

exports[`icon with color 1`] = `
"<Frame width={48} height={48}>
  <Box
    w=\\"100%\\"
    h=\\"100%\\"
    bgColor=\\"#60a5fa\\"
    style={{
      maskImage:
        'url(https://api.iconify.design/material-symbols/menu.svg) center / contain no-repeat',
      WebkitMask:
        'url(https://api.iconify.design/material-symbols/menu.svg) center / contain no-repeat',
    }}
  />
</Frame>;
"
`;

exports[`link with icon 1`] = `
"<Frame width={100} height={40}>
  <Link
    fontWeight={500}
    textColor=\\"dodgerblue\\"
    textDecorationColor=\\"dodgerblue\\"
  >
    Test
  </Link>
</Frame>;
"
`;

exports[`text default 1`] = `
"<Frame width={100} height={40}>
  <Text textAlign=\\"center\\" className=\\"text-red-500\\">
    Hello world
  </Text>
</Frame>;
"
`;

exports[`text escape jsx 1`] = `
"<Frame width={100} height={40}>
  <Text>{'Hello \\"\\\\'<{test}>\\\\'\\" world'}</Text>
</Frame>;
"
`;
